File: Sc.Models/Dtos/Suppliers/Search/SupplierOfferingGroup/SmartSpecializationGroupDto.cs
using Sc.Models.Entities.Base;

namespace Sc.Models.Dtos.Suppliers.Search.SupplierOfferingGroup
{
    public class SmartSpecializationGroupDto : Entity
    {
        public string Code { get; set; }
        public string Name { get; set; }
        public string NameAlt { get; set; }
        public int ViewOrder { get; set; }

        public int OfferingsCount { get { return Offerings.Count; } }
        public List<SmartSpecializationOfferingGroupDto> Offerings { get; set; } = new List<SmartSpecializationOfferingGroupDto>();
    }
}

File: WebApplication/src/suppliers/dtos/search/supplier-offering-group/smart-specialization-group.dto.ts
import { SmartSpecializationOfferingGroupDto } from "./smart-specialization-offering-search.dto";

export class SmartSpecializationGroupDto {
    id: number;

    code: string;
    name: string;
    nameAlt: string;
    viewOrder: number;

    offeringsCount: number;
    offerings: SmartSpecializationOfferingGroupDto[] = [];
}